sources/api/migration/migrations/v1.12.0/add-k8s-autoscaling-warm-pool-setting/src/main.rs

```diff
@@ -0,0 +1,25 @@
+#![deny(rust_2018_idioms)]
+
+use migration_helpers::common_migrations::AddPrefixesMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a new setting prefix for configuring autoscaling in k8s variants.
+/// Remove the whole `settings.autoscaling` prefix if we downgrade.
+fn run() -> Result<()> {
+    migrate(AddPrefixesMigration(vec![
+        "settings.autoscaling",
+        "services.autoscaling-warm-pool",
+        "configuration-files.warm-pool-wait-toml",
+    ]))
+}
```

Comment:
This needs to be conditionalized on the `aws-k8s` family, otherwise it will run on downgrade for `aws-ecs-1` and remove `settings.autoscaling` (which has been in that variant since 1.8.0.

`migrations/archived/v1.8.0/pki-affected-services` has an example of doing this; there are two parts, one in `build.rs` to set the variables, and another in `main.rs` to consume them.

It shouldn't hurt to do the same to the `metadata` migration but it's not strictly necessary since `aws-ecs-1` doesn't have that metadata.